packages/components/src/GridLayout/Grid.component.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'react-grid-layout/css/styles.css';
+import { Responsive, WidthProvider } from 'react-grid-layout';
+import Tile from './Tile';
+import { SKELETON_TILE_CONF } from './Tile/Skeleton/SkeletonTile.component';
+import './Grid.scss';
+
+// eslint-disable-next-line new-cap
+const ResponsiveGridLayout = WidthProvider(Responsive);
+const MARGIN = 20;
+
+const columns = {
+	xs: 2,
+	s: 4,
+	m: 6,
+	l: 12,
+};
+
+const breakpoints = {
+	xs: 479,
+	s: 768,
+	m: 1024,
+	l: 1366,
+};
+
+function onLayoutChange(layoutChangeCallback) {
+	return layout => {
+		if (layoutChangeCallback) {
+			layoutChangeCallback(layout);
+		}
+	};
+}
+
+function onBreakpointChange(onBreakpointChangeCallback) {
+	return (breakpoint, cols) => {
+		if (onBreakpointChangeCallback) {
+			onBreakpointChangeCallback(breakpoint, cols);
+		}
+	};
+}
+
+function onDragStop(onDragStopCallback) {
+	return (layout, oldItem, newItem, placeHolder, event) => {
+		if (onDragStopCallback) {
+			onDragStopCallback(layout, oldItem, newItem, placeHolder, event);
+		}
+	};
+}
+
+function onResizeStop(onResizeStopCallback) {
+	return (layout, oldItem, newItem, placeHolder, event) => {
+		if (onResizeStopCallback) {
+			onResizeStopCallback(layout, oldItem, newItem, placeHolder, event);
+		}
+	};
+}
+
+function Grid({
+	children,
+	isResizable = false,
+	layoutChangeCallback,
+	onBreakpointChangeCallback,
+	onDragStopCallback,
+	onResizeStopCallback,
+	isLoading = false,
+	skeletonConfiguration,
+}) {
+	return (
+		<ResponsiveGridLayout
+			className="layout"
+			onLayoutChange={onLayoutChange(layoutChangeCallback)}
+			onDragStop={onDragStop(onDragStopCallback)}
+			onResizeStop={onResizeStop(onResizeStopCallback)}
+			onBreakpointChange={onBreakpointChange(onBreakpointChangeCallback)}
```

Comment:
You don't need those higher order functions
```javascript
function noOp = () => {};

function Grid({
	children,
	isResizable = false,
	onLayoutChange = noOp,
	onBreakpointChange = noOp,
	onDragStop = noOp,
	onResizeStop = noOp,
	isLoading = false,
	skeletonConfiguration,
}) {
	return (
		<ResponsiveGridLayout
			className="layout"
			onLayoutChange={onLayoutChange}
			onDragStop={onDragStop}
			onResizeStop={onResizeStop}
			onBreakpointChange={onBreakpointChange}
```

Reply:
yup did it at first cause i thought i might add some behaviour inside UI then call my host app function. but didn't have to in the end.